fvm/evm/stdlib/contract.go

```diff
@@ -97,6 +99,55 @@ func (e abiDecodingError) Error() string {
 	return b.String()
 }
 
+func calculateComputation(
+	inter *interpreter.Interpreter,
+	values *interpreter.ArrayValue,
+	evmAddressTypeID common.TypeID,
+) uint {
+	computation := uint(0)
+
+	values.Iterate(inter, func(element interpreter.Value) (resume bool) {
+		switch value := element.(type) {
```

Comment:
Maybe refactor this inner function into a separate function for just `interpreter.Value`

Reply:
Ah, I see now: The "outer-most" array of all values that are encoded is not itself a dynamic array, only inner arrays are and need to be metered that way. Yeah, the function looks good now 👍 

When the function used to sum the computation instead of reporting it immediately, I was mostly just concerned that the `computation` would be potentially overwritten, and returning it from the function for a single value is less error prone. That's not a concern anymore now.

I factored out the repeated computation kind, which should always be the sane, in a952d6633c6351e418fc8fe4581546d6649a817e